google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/AuthorizedView.java

```diff
@@ -0,0 +1,122 @@
+ * except that visibility is restricted to the subset of the Table that the AuthorizedView
+ * represents.
+ */
+public final class AuthorizedView {
+  private final com.google.bigtable.admin.v2.AuthorizedView proto;
+
+  /**
+   * Wraps the protobuf. This method is considered an internal implementation detail and not meant
+   * to be used by applications.
+   */
+  @InternalApi
+  public static AuthorizedView fromProto(
+      @Nonnull com.google.bigtable.admin.v2.AuthorizedView proto) {
+    return new AuthorizedView(proto);
+  }
+
+  private AuthorizedView(@Nonnull com.google.bigtable.admin.v2.AuthorizedView proto) {
+    Preconditions.checkNotNull(proto);
+    Preconditions.checkArgument(!proto.getName().isEmpty(), "AuthorizedView must have a name");
+    Preconditions.checkArgument(
+        proto.hasSubsetView(), "AuthorizedView must have a subset_view field");
+    this.proto = proto;
+  }
+
+  /** Gets the authorized view's id. */
+  public String getId() {
+    // Constructor ensures that name is not null.
+    AuthorizedViewName fullName = AuthorizedViewName.parse(proto.getName());
+
+    //noinspection ConstantConditions
+    return fullName.getAuthorizedView();
+  }
+
+  /** Gets the id of the table that owns this authorized view. */
+  public String getTableId() {
+    // Constructor ensures that name is not null.
+    AuthorizedViewName fullName = AuthorizedViewName.parse(proto.getName());
+
+    //noinspection ConstantConditions
+    return fullName.getTable();
+  }
+
+  /** Returns whether this authorized view is deletion protected. */
+  public boolean isDeletionProtected() {
+    return proto.getDeletionProtection();
+  }
+
+  /** Gets the type of this authorized view, which currently can only be a subset view. */
+  public AuthorizedViewType getAuthorizedViewType() {
+    if (proto.hasSubsetView()) {
+      return SubsetView.fromProto(proto.getSubsetView());
+    } else {
+      // Should never happen because the constructor verifies that one must exist.
+      throw new IllegalStateException("This AuthorizedView doesn't have a valid type specified");
+    }
+  }
+
+  /**
+   * Creates the request protobuf. This method is considered an internal implementation detail and
+   * not meant to be used by applications.
+   */
+  @InternalApi
+  public com.google.bigtable.admin.v2.AuthorizedView toProto() {
+    return proto;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    AuthorizedView that = (AuthorizedView) o;
+    return Objects.equal(proto, that.proto);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(proto);
+  }
+
+  /**
+   * Represents a subset of a Table. Please check the implementations of this interface for more
+   * details.
+   */
+  public interface AuthorizedViewType {}
```

Comment:
I can't remember where we landed on this, but this should probably be marked as `@InternalApi` or `@InternalExtensionOnly`

Reply:
We have a `getAuthorizedViewType()` public method (as below) to allow users to inspect what's in their authorized views. This method returns an `AuthorizedViewType` object.

```
/** Gets the type of this authorized view, which currently can only be a subset view. */
public AuthorizedViewType getAuthorizedViewType() {
  if (proto.hasSubsetView()) {
    return SubsetView.fromProto(proto.getSubsetView());
  } else {
    // Should never happen because the constructor verifies that one must exist.
    throw new IllegalStateException("This AuthorizedView doesn't have a valid type specified");
  }
}
```

I'm not familiar with Java annotations. In this case, can it still be marked as `@InternalApi` or `@InternalExtensionOnly`?